GlCanvas.js

```diff
@@ -0,0 +1,187 @@
+(function() {
+function GlPlotter() {
+    this.pluginName = 'glplotter';
+    this.pluginVersion = '0.2';
+
+    /**
+     * Initialize a webgl layer
+     */
+    this.init = function(plot) {
+        var defaultPlotOffset = {
+            left: 0,
+            right: 0,
+            top: 0,
+            bottom: 0
+        };
+
+        var textures = [],
+            materials = [],
+            geometries = [];
+
+        plot.hooks.processOptions.push(processOptions);
+
+        function processOptions(plot, options) {
+            var container = plot.getPlaceholder(),
+                webglsurface = plot.getWebGlSurface(),
+                canvas = plot.getWebGlCanvas(),
+                width = webglsurface.width,
+                height = webglsurface.height,
+                renderer = new THREE.WebGLRenderer({ canvas: canvas, antialias: false, alpha: true }),
+                mainScene = new THREE.Scene(),
+                camera = new THREE.OrthographicCamera(webglsurface.width / 2, -webglsurface.width / 2, -webglsurface.height / 2, webglsurface.height / 2, 0.1, 1000),
+                cameraFocus = new THREE.Vector3(width / 2, height / 2, 1000);
+
+            // Setup the THREE.OrthographicCamera
+            camera.aspect = width / height;
+            camera.position.set(width / 2, height / 2, 0);
+            camera.lookAt(cameraFocus);
+            camera.updateMatrixWorld();
+            camera.updateProjectionMatrix();
+
+            // Setup the THREE.WebGLRenderer
+            renderer.setPixelRatio(webglsurface.pixelRatio);
+            renderer.setSize(webglsurface.width, webglsurface.height, true);
+            renderer.autoClear = true;
```

Comment:
In case that, for whatever reason, the drawing can't keep up to 60fps, will the canvas be cleared? Is it possible for the user to see an empty plot for a few ms because the render() wasn't called meantime?

Reply:
renderer.autoClear defines whether the renderer should automatically clear its output before rendering a frame.

Reply:
and the answer is no, because it will autoclear only when renderer.render(scene, camera) is called.